Samples/custom-transform-adapter/server.go

```diff
@@ -0,0 +1,284 @@
+
+			if err != nil {
+				respondError(logger, w, http.StatusBadRequest, fmt.Sprintf("Payload transformation failed: %s", err.Error()))
+				return
+			}
+		} else {
+			transformedPayload = jsonBody
+		}
+
+		if err := decodeDateTimeField(&transformedPayload, "creationTimeUtc"); err != nil {
+			respondError(logger, w, http.StatusBadRequest, fmt.Sprintf("Failed to parse \"creationTimeUtc\": %s", err.Error()))
+			return
+		}
+
+		// Convert transformation output to Autorest typed input.
+		var bridgePayload bridge.MessageBody
+		if err := mapstructure.Decode(transformedPayload, &bridgePayload); err != nil {
+			respondError(logger, w, http.StatusBadRequest, fmt.Sprintf("Failed to transform payload to expected Device Bridge format: %s", err.Error()))
+			return
+		}
+
+		bridgeClient := adapter.GetBridgeClient()
+
+		// Extracts the API key from the query parameter or header.
+		var apiKey string
+		if message.AuthQueryParam != "" {
+			values, ok := r.URL.Query()[message.AuthQueryParam]
+
+			if !ok || len(values) < 1 {
+				respondError(logger, w, http.StatusBadRequest, fmt.Sprintf("Expected auth query parameter \"%s\" to be defined", message.AuthQueryParam))
+				return
+			}
+
+			apiKey = values[0]
+		} else if message.AuthHeader != "" {
+			apiKey = r.Header.Get(message.AuthHeader)
+		} else {
+			respondError(logger, w, http.StatusBadRequest, "No auth method specified")
+			return
+		}
+
+		bridgeClient.SetAuthorizer(autorest.NewAPIKeyAuthorizerWithHeaders(map[string]interface{}{
+			"x-api-key": apiKey,
+		}))
+
+		var deviceId string
+		if message.DeviceIdBodyQueryId != "" {
+			var queriedDeviceId interface{}
+			var err error
+			if queriedDeviceId, err = adapter.Engine.Execute(message.DeviceIdBodyQueryId, jsonBody); err != nil {
+				respondError(logger, w, http.StatusBadRequest, fmt.Sprintf("Device Id body query failed: %s", err.Error()))
+				return
+			}
+
+			var ok bool
+			if deviceId, ok = queriedDeviceId.(string); !ok || deviceId == "" {
+				respondError(logger, w, http.StatusBadRequest, "Expected result from device Id body query to be string")
+				return
+			}
+		} else if message.DeviceIdPathParam != "" {
+			var ok bool
+			if deviceId, ok = mux.Vars(r)[message.DeviceIdPathParam]; !ok {
+				respondError(logger, w, http.StatusBadRequest, fmt.Sprintf("Expected device Id in \"%s\" path parameter", message.DeviceIdPathParam))
+				return
+			}
+		} else {
+			respondError(logger, w, http.StatusBadRequest, "No device Id specified")
+			return
+		}
+
+		bridgeClient.SetRetryAttempts(1) // Don't retry (the Bridge already has internal retries)
+
+		if bridgeResponse, err := bridgeClient.SendMessage(r.Context(), deviceId, &bridgePayload); err != nil {
+			// We return the Bridge status code if we have it
+			var responseStatusCode int
+			if bridgeResponse != (autorest.Response{}) {
+				responseStatusCode = bridgeResponse.StatusCode
+			} else {
+				responseStatusCode = http.StatusInternalServerError
+			}
```

Comment:
this is more idiomatic go:

```suggestion
			responseStatusCode := http.StatusInternalServerError
			if bridgeResponse != autorest.Response{} {
				responseStatusCode = bridgeResponse.StatusCode
			}
```